[Azure Pipelines] Fix `wpt test-jobs` failures

stephenmcgruer opened the pull request:
For some reason, Azure Pipelines is not consistently fetching `origin/master`, which
results in a failing `wpt test-jobs` command. Even worse, we didn't catch the failure
because our script pipes the outcome of `./wpt test-jobs` and doesn't set `pipefail`!

This PR works around the `origin/master` problem by explicitly fetching it, and also
adds the appropriate bash `set` command to fail on pipes and other errors.

Fixes #26448

stephenmcgruer committed:
[Azure Pipelines] Set `-e` and `-o pipefail` for decision task

Fixes #26448

stephenmcgruer commented:
Good news - the `wpt test-jobs` failure is now detected.

Bad news - `wpt test-jobs` is failing :D

```
Traceback (most recent call last):
  File "./wpt", line 21, in <module>
    wpt.main()
  File "/home/vsts/work/1/s/tools/wpt/wpt.py", line 173, in main
    rv = script(*args, **kwargs)
  File "/home/vsts/work/1/s/tools/ci/jobs.py", line 143, in run
    paths = get_paths(**kwargs)
  File "/home/vsts/work/1/s/tools/ci/jobs.py", line 93, in get_paths
    revish = "%s..HEAD" % branch_point()
  File "/home/vsts/work/1/s/tools/wpt/testfiles.py", line 110, in branch_point
    merge_base = git("merge-base", "HEAD", "origin/master")
  File "/home/vsts/work/1/s/tools/manifest/utils.py", line 60, in gitfunc
    return subprocess.check_output(full_cmd, cwd=path, stderr=subprocess.STDOUT).decode('utf8')
  File "/usr/lib/python2.7/subprocess.py", line 223, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '[u'git', 'merge-base', 'HEAD', 'origin/master']' returned non-zero exit status 128
```

Hexcles commented on .azure-pipelines.yml:
```diff
@@ -79,6 +79,7 @@ jobs:
   steps:
   - template: tools/ci/azure/checkout.yml
   - script: |
+      set -e; set -o pipefail
```
```suggestion
      set -euo pipefail
```

Let's do `set -u` by the way. And they can be in a single line.

Hexcles reviewed (approved):
LGTM % a nit

stephenmcgruer committed:
Add debug logging

stephenmcgruer committed:
Try this

stephenmcgruer committed:
Revert "Add debug logging"

This reverts commit 308c5ad91d9d3496650b1117cbc772a719862944.

stephenmcgruer commented:
Added debug logging, the error is : `fatal: Not a valid object name origin/master`. I wonder if Azure Pipelines... somehow does some sort of very shallow fetch and doesn't grab the remote branches?

For now I've added an explicit fetch, but open to how better to do this. (Will also need to rename this PR before landing)

Hexcles commented:
Looks like Azure Pipelines use two different kinds of checkout commands:

This works fine:
https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=57129&view=logs&j=3a8a797a-cea8-5b38-08dd-c96687b91c17&t=ce64449e-0287-561d-7a51-6630f4993bc5&l=21
`git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=50 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/19706/merge:refs/remotes/pull/19706/merge`

This doesn't:
https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=57177&view=logs&j=3a8a797a-cea8-5b38-08dd-c96687b91c17&t=ce64449e-0287-561d-7a51-6630f4993bc5&l=21
`git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=50 origin +0f4fffebbc9afa54bd5e2aef85c14ab0b144465d:refs/remotes/origin/0f4fffebbc9afa54bd5e2aef85c14ab0b144465d`

It's unclear to me what's the difference between the two (they're both PR runs), but it's a rabbit hole that I don't plan to go down today.

Hexcles commented on .azure-pipelines.yml:
```diff
@@ -79,6 +79,8 @@ jobs:
   - template: tools/ci/azure/checkout.yml
   - script: |
+      set -euo pipefail
+      git fetch origin
```
Maybe
```suggestion
      git fetch --depth 50 --quiet origin master
```
to avoid fetching everything (albeit it seems to only take <30s anyway) and make the output less verbose

stephenmcgruer committed:
Update .azure-pipelines.yml

Co-authored-by: Robert Ma <robertma@chromium.org>
